### PR TITLE
Refactor the project setup and achieve feature parity with native JudoKit.

### PR DESCRIPTION
**Pull request by stefan-tudor**
- Moving the project to Yarn Workspaces
- Recommendation feature implementation
- Apple Pay recurring configuration
- Should ask for `CSC` and `Cardholder Name`
- Fastlane setup
- ReactNative updated to `0.73.2`
- Various improvements across the project

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,233 +1,96 @@
...
-def DEFAULT_MIN_SDK_VERSION = 21
-def DEFAULT_TARGET_SDK_VERSION = 33
+  repositories {
+    google()
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath "com.android.tools.build:gradle:7.2.1"
+    // noinspection DifferentKotlinGradleVersion
```
Comment (RafalOzogJudopay) [inline android/build.gradle:12]: Do you maybe remember what was this commented line about, Stefan?

File: android/src/main/AndroidManifestNew.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+</manifest>
```
Comment (RafalOzogJudopay) [inline android/src/main/AndroidManifestNew.xml:2]: Do we need this AndroidManifestNew along with AndroidManifest?

File: android/src/main/java/com/judopay/judokit/reactnative/Extensions.kt [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+  } else {
+    null
+  }
+
+internal fun ReadableMap?.getOptionalArray(key: String): ReadableArray? =
+  if (hasKey(key)) {
+    this?.getArray(key)
+  } else {
+    null
+  }
```
Comment (RafalOzogJudopay) [inline android/src/main/java/com/judopay/judokit/reactnative/Extensions.kt:58]: I think we can simplify these if-elses into 1-liners to make it shorter ;)

**Review by RafalOzogJudopay — APPROVED** [review]
That's a big PR, 300 files! I did go through this roughly, but to be honest it's not easy to grasp all of the changes just by looking at it.

I remember we had a call about new setup, so I will try to setup it locally as well quite soon, to test how it works at my side after the changes.

**Review by RafalOzogJudopay — APPROVED** [review]

File: android/src/main/AndroidManifestNew.xml (hunk 2 above)
Comment (stefan-tudor) [reply android/src/main/AndroidManifestNew.xml:2]: Yes,
Screenshot 2024-02-13 at 09 53 18

**Pull request merged**